custom_libs/subliminal_patch/providers/jimaku.py

```diff
@@ -0,0 +1,421 @@
+                '\u4E00' <= char <= '\u9FFF' or  # Kanji (CJK Unified Ideographs)
+                '\uFF66' <= char <= '\uFF9D'):   # Half-width Katakana
+                return True
+        return False
+    
+    @staticmethod
+    def _is_archive(archive: bytes):
+        archive_stream = io.BytesIO(archive)
+        
+        if rarfile.is_rarfile(archive_stream):
+            logger.debug("Identified rar archive")
+            return rarfile.RarFile(archive_stream)
+        elif zipfile.is_zipfile(archive_stream):
+            logger.debug("Identified zip archive")
+            return zipfile.ZipFile(archive_stream)
+        else:
+            logger.debug("Doesn't seem like an archive")
+            return None
+        
+    @staticmethod
+    def _subtitle_sorting_key(file):
+        name = file["name"].lower()
+        is_srt = name.endswith('.srt')
+        
+        # Usually netflix > webrip > amazon, with the rest having the lowest priority
+        sub_sources = ["netflix", "webrip", "amazon"]
+        priority = len(sub_sources)
+        for index, source in enumerate(sub_sources):
+            if source in name:
+                priority = index
+                break
+        
+        return (not is_srt, priority, file["name"])
+    
+    @cache
+    def _get_jimaku_response(self, url_path):
+        url = f"{self.api_url}/{url_path}"
+        
+        retry_count = 0
+        while retry_count < self.api_ratelimit_backoff_limit:
+            retry_count += 1
+            response = self.session.get(url, timeout=10)
+            
+            if response.status_code == 429:
+                api_reset_time = float(response.headers.get("x-ratelimit-reset-after", 5))
+                reset_time = self.api_ratelimit_max_delay if api_reset_time > self.api_ratelimit_max_delay else api_reset_time
+                
+                logger.warning(f"Jimaku ratelimit hit, waiting for '{reset_time}' seconds ({retry_count}/{self.api_ratelimit_backoff_limit} tries)")
+                time.sleep(reset_time)
+            else:
+                response.raise_for_status()
+            
+            data = response.json()
+            logger.debug(f"Length of response on {url}: {len(data)}")
+            if len(data) == 0:
+                logger.error(f"Jimaku returned no items for our our query: {url_path}")
+                return None
+            else:
+                return data
+            
+        # Max retries exceeded
+        raise TooManyRequests(f"Jimaku ratelimit max backoff limit of {self.api_ratelimit_backoff_limit} reached, aborting.")
+    
+    @staticmethod
+    @cache
+    def _webrequest_with_cache(url, headers=None):
+        response = requests.get(url, headers=headers)
+        response.raise_for_status()
+        
+        try:
+            return response.json()
+        except:
+            return response.content
+
+    def _get_tvdb_anidb_mapping(self, video, tvdbid):
```

Comment:
You should use the existing refiner for anidb

Reply:
Switched to refiner in [`aa18f74`](https://github.com/morpheus65535/bazarr/pull/2505/commits/aa18f7407a2801a792d17c90e710a50b5c9c5f2c).